common/djangoapps/util/course.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utility methods related to course
+"""
+import logging
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+import re
+
+log = logging.getLogger(__name__)
+
+
+def get_lms_link_for_about_page(request, course_key):
+    """
+    Returns the url to the course about page from the location tuple.
+    """
+
+    assert isinstance(course_key, CourseKey)
+
+    if settings.FEATURES.get('ENABLE_MKTG_SITE', False):
+        if not hasattr(settings, 'MKTG_URLS'):
+            log.exception("ENABLE_MKTG_SITE is True, but MKTG_URLS is not defined.")
+            return None
+
+        marketing_urls = settings.MKTG_URLS
+
+        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
+        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
+        about_base = marketing_urls.get('ROOT', None)
+
+        if about_base is None:
+            log.exception('There is no ROOT defined in MKTG_URLS')
+            return None
+
+        # Strip off https:// (or http://) to be consistent with the formatting of LMS_BASE.
+        about_base = re.sub(r"^https?://", "", about_base)
+
+    elif hasattr(settings, "LMS_BASE"):
```

Comment:
This utility already existed at "cms/djangoapps/contentstore/utils.py"; I have only moved it to common to make use of it for both lms and cms.
For other apps (e.g. cms) "LMS_BASE" is configured in settings while for lms itself its not; so I have introduced this new else clause at the end to get base url from request.
`
if settings.FEATURES.get('ENABLE_MKTG_SITE', False):
         # do some work
elif settings.LMS_BASE is not None:
       # do some work
**else:
 +        about_base = request.META['HTTP_HOST']**
`
@cahrens @nasthagiri Do we have any common place for all apps base_urls that we can use to avoid this check? @verdanmahmood mentioned ENV_TOKENS in envs/aws.py; can we make use of it?